pkg/daemon/daemon.go

```diff
@@ -214,23 +218,36 @@ func (dn *Daemon) addProfileConfig(socketPath string, nodeProfile *ptpv1.PtpProf
 		nodeProfile.Ptp4lOpts = &opts
 	}
 
-	if nodeProfile.Interface == nil || *nodeProfile.Interface == "" {
-		// We need to copy this to another variable because is a pointer
+	output := &ptp4lConf{}
+	err := output.populatePtp4lConf(nodeProfile.Ptp4lConf)
+	if err != nil {
+		return err
+	}
+
+	if nodeProfile.Interface != nil && *nodeProfile.Interface != "" {
+		ifaceSection := fmt.Sprintf("[%s]", *nodeProfile.Interface)
+		output.sections[ifaceSection] = *new(ptp4lConfSection)
+	} else {
 		iface := string("")
 		nodeProfile.Interface = &iface
 	}
 
-	config := fmt.Sprintf("%s\nuds_address %s\nmessage_tag [%s]",
-		*nodeProfile.Ptp4lConf,
-		socketPath,
-		*nodeProfile.Interface)
-	nodeProfile.Ptp4lConf = &config
+	section := output.sections["[global]"]
+	section.options["message_tag"] = fmt.Sprintf("[%s]", configFile)
+	section.options["uds_address"] = socketPath
+	output.sections["[global]"] = section
+
+	output.renderPtp4lConf()
```

Comment:
this function can return 2 strings ptp4lConf and the interface list.

There is no need have them as fields in the struct of the ptp4lConf